src/init.cpp

```diff
@@ -2036,29 +1998,18 @@ bool AppInitMain(boost::thread_group& threadGroup, CScheduler& scheduler)
 
         strDBName = "mncache.dat";
         uiInterface.InitMessage(_("Loading masternode cache..."));
-        CFlatDB<CMasternodeMan> flatdb1(strDBName, "magicMasternodeCache");
-        if(!flatdb1.Load(mnodeman)) {
+        CFlatDB<CMasternodeMetaMan> flatdb1(strDBName, "magicMasternodeMetaCache");
```

Comment:
Is altering magic necessary? This will require user to _manually_ delete the file on wallet upgrade which could be annoying. Altering "version" should already do the job and clear all the previous data.

Reply:
ah didn't notice the extra handling of the magic string and assumed it's only the string compare in the serialization methods. Will change this later.

Reply:
Yep, the purpose of the magic string is to ensure that this is exactly the kind of file we expected and not some renamed wallet.dat for example (in which case user will have to manage things manually and rename it to smth else).